src/drive/web/modules/navigation/FileExplorer.jsx

```diff
@@ -176,11 +178,19 @@ const mapDispatchToProps = (dispatch, ownProps) => ({
 })
 
 export const handleFileOpen = (file, availableOffline, props, dispatch) => {
+  const redirectToViewer = () => {
+    const viewPath = props.location.pathname
+    props.router.push(`${viewPath}/file/${file.id}`)
+  }
   if (availableOffline) {
     return dispatch(openLocalFile(file))
   }
 
   if (models.file.isNote(file)) {
+    //This is a tmp hack since Notes doesn't handle sharing properly.
+    if (!isNoteMine(file, props.client)) {
+      return redirectToViewer()
+    }
```

Comment:
There are 3 calls to `redirectToViewer ()` and the `return` is a bit misleading, since it's only there to stop the function execution.

It could be rewritten as

```
const isNote = models.file.isNote(file)
const isMyNote = isNote && isNoteMine(file, props.client)

if (isMyNote) { … } else { redirectToViewer() }
```

Reply:
Thanks for the suggestion. Need to call `redirectToViewer` twice but much more cleaner :)